### PR TITLE
Setup connection simplest spec

### PR DESCRIPTION
**Pull request by alex-dukhno**

### CODE REVIEW

File: tck-core/src/main/kotlin/io/rsocket/frame/SetupFrame.kt [hunk 1]
```diff
@@ -9,11 +9,17 @@ data class SetupFrame(
...
     val resumeToken: ResumeToken?,
-    val metadataMimeType: MimeType,
-    val dataMimeType: MimeType,
-    val payload: Payload
+    val metadataMimeType: MimeType?,
+    val dataMimeType: MimeType?,
+    val payload: Payload?
 ) : Frame<SetupFlags>(FrameType.SETUP) {
 
+    constructor(
```
Comment (whyoleg) [inline tck-core/src/main/kotlin/io/rsocket/frame/SetupFrame.kt]: null can be used as default parameter in primary constructor

File: tck-core/src/main/kotlin/io/rsocket/frame/SetupFrame.kt [hunk 2]
```diff
@@ -23,14 +29,25 @@ data class SetupFrame(
                 writeShort(it.length.toInt())
                 it.token.preview { this@headerBuffer.writeBytes(this) }
             }
-            writeByte(metadataMimeType.length.toInt())
-            writeUtf8(metadataMimeType.text)
-            writeByte(dataMimeType.length.toInt())
-            writeUtf8(dataMimeType.text)
-            payload.metadata?.length?.let(this::writeLength)
+            if (metadataMimeType != null) {
```
Comment (whyoleg) [inline tck-core/src/main/kotlin/io/rsocket/frame/SetupFrame.kt]: As I remember mime types are required
so, need to write at least one byte with 0 value

File: tck-spec/build.gradle [hunk 3]
```diff
@@ -3,6 +3,9 @@ dependencies {
     implementation("org.spekframework.spek2:spek-dsl-jvm:$spek_version")
     testRuntimeOnly("org.spekframework.spek2:spek-runner-junit5:$spek_version")
     testRuntimeOnly("org.jetbrains.kotlin:kotlin-reflect:$kotlin_version")
+    testImplementation("io.cucumber:cucumber-java:5.6.0")
```
Comment (whyoleg) [inline tck-spec/build.gradle]: with kotlin, better to use `cucumber-java8` with lambda syntax

File: tck-spec/src/test/kotlin/io/rsocket/spec/StepDefinitions.kt [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    @Then("server does not close connection")
+    fun finishTest() {
+        assert(expected == actual)
+    }
+}
+
+class Server(private val port: Int) {
+    @Throws(Exception::class)
```
Comment (whyoleg) [inline tck-spec/src/test/kotlin/io/rsocket/spec/StepDefinitions.kt]: `Throws` annotation isn't needed, if you not use both kotlin and java in one project

File: tck-spec/src/test/kotlin/io/rsocket/spec/StepDefinitions.kt [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+import io.rsocket.frame.shared.KeepAlive
+import io.rsocket.frame.shared.Version
+import java.util.concurrent.TimeUnit.MILLISECONDS
+import kotlin.time.toDuration
+
+class StepDefinitions {
+    private val initializer = ServerChannelInitializer()
+    private lateinit var server: Server
+    private lateinit var expected: SetupFrame
+    private lateinit var actual: SetupFrame
```
Comment (whyoleg) [inline tck-spec/src/test/kotlin/io/rsocket/spec/StepDefinitions.kt]: actual not assigned?

**Review by whyoleg — COMMENTED** [review]
minor `kotlin` style related things

File: tck-core/src/main/kotlin/io/rsocket/frame/SetupFrame.kt (hunk 2 above)
Comment (alex-dukhno) [reply tck-core/src/main/kotlin/io/rsocket/frame/SetupFrame.kt]: @OlegDokuka I've not found if mime type is required or not in RSocket spec. 
My question is should spec says explicitly is mime type required during connection establishment?

**Review by OlegDokuka — APPROVED** [review]
LGTM

**Pull request merged**